app/screens/edit_post/post_input/index.tsx

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {forwardRef, useCallback, useImperativeHandle, useRef} from 'react';
+import {useIntl} from 'react-intl';
+import {KeyboardType, Platform, TextInput, useWindowDimensions, View} from 'react-native';
+
+import {useTheme} from '@context/theme';
+import {changeOpacity, getKeyboardAppearanceFromTheme, makeStyleSheetFromTheme} from '@utils/theme';
+import {typography} from '@utils/typography';
+
+const getStyleSheet = makeStyleSheetFromTheme((theme) => ({
+    input: {
+        color: theme.centerChannelColor,
+        padding: 15,
+        textAlignVertical: 'top',
+        ...typography(),
+    },
+    inputContainer: {
+        borderTopWidth: 1,
+        borderBottomWidth: 1,
+        borderTopColor: changeOpacity(theme.centerChannelColor, 0.1),
+        borderBottomColor: changeOpacity(theme.centerChannelColor, 0.1),
+        backgroundColor: theme.centerChannelBg,
+        marginTop: 2,
+    },
+}));
+
+export type PostInputRef = {
+    focus: () => void;
+}
+type PostInputProps = {
+    keyboardType: KeyboardType;
+    message: string;
+    hasError: boolean;
+    onPostSelectionChange: (curPos: number) => void;
+    onChangeText: (text: string) => void;
+}
+const PostInput = forwardRef<PostInputRef, PostInputProps>(({keyboardType, message, onChangeText, onPostSelectionChange, hasError}: PostInputProps, ref) => {
+    const theme = useTheme();
+    const intl = useIntl();
+    const {height} = useWindowDimensions();
+    const inputRef = useRef<TextInput>(null);
+
+    useImperativeHandle(ref, () => ({
+        focus: () => inputRef.current?.focus(),
+    }), [inputRef]);
```

Comment:
do you need this?

Reply:
There is a behaviour from v1 that focusses the text input back when there is an error.